apps/okreads-e2e/src/specs/reading-list.spec.ts

```diff
@@ -17,4 +17,53 @@ describe('When: I use the reading list feature', () => {
       )
     );
   });
+
+  it('Then: I should revert book Added to reading list', async () => {
+    await browser.get('/');
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement($('tmo-root'), 'okreads')
+    );
+
+    const form = await $('form');
+    const input = await $('input[type="search"]');
+    await input.sendKeys('Python');
+    await form.submit();
+
+    const addBooksToListButton = await $$('[data-testing="add-to-list"] [ng-reflect-disabled="false"]');
+    addBooksToListButton[0].click();
+
+    const snackbarButton =  await $$('snack-bar-container .mat-button');
+
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement(snackbarButton[0], 'Undo')
+    );
+    snackbarButton[0].click();
+  });
+
+  it('Then: I should revert book removed from the reading list', async () => {
+    await browser.get('/');
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement($('tmo-root'), 'okreads')
+    );
+
+    const readingListToggle = await $('[data-testing="toggle-reading-list"]');
+    await readingListToggle.click();
+
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement(
+        $('[data-testing="reading-list-container"]'),
+        'My Reading List'
+      )
+    );
+
+    const readingListBooksButton = await $$('[data-testing="reading-list"]');
+    readingListBooksButton[0].click();
+
+    const snackbarButton =  await $$('snack-bar-container .mat-button');
+
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement(snackbarButton[0], 'Undo')
+    );
+    snackbarButton[0].click();
```

Comment:
You are not actually testing here if the undo action succeeded, though! You are only clicking the snackbar button, but are not observing if it worked or not!

Reply:
Updated changes, please review. Thanks!